MyModels.py

```diff
@@ -0,0 +1,158 @@
+class BaseModel:
+    def __init__(self):
+        self.list = []
+        self.id_index = {}
+        
+    def add(self, entry):
+        
+        self.list.append(entry)
+    
+#     def find_index(self,entry):
+#         for i, item in self.list:
+#             if item==entry:
+#                 return i
+            
+#     def find_val(self,index):
+#         return self.list[index]
+
+#####################
+
+    def find_index(self,id):
+        return self.id_index[id]
+    
+    def find_val(self,id):
+        index = self.find_index(id)
+        return self.list[index]
+
+
+###################
+    def GetAll(self):
+        return self.list
+    
+    def delete(self, entry):
+        
+        self.list.remove(entry)
+    
+    def update(self,id, new_entry):
+        
+        index = self.id_index[id]
+        self.list[index] = new_entry
+        
+    
+
+class User(BaseModel):
+    
+    def __init__(self):
+        self.email_userid = {}
+        self.userid = 0
+        super().__init__()
+        
+    
+    def register(self, client_name, email, password):
+        entry = {'client_name' : client_name,
+                'email': email,
+                'password':password,
+                'userid': self.userid }
+        self.add(entry = entry)
+        self.id_index[self.userid] = len(self.list) - 1
+        self.email_userid[email] = self.userid
+        
+        self.userid += 1
+        
+#     def find_index(self,userid):
+#         return self.id_index[userid]
+    
+#     def find_val(self,userid):
+#         index = self.find_index(userid)
+#         return self.list[index]
+    
+    def validate(self,email,password):
+        
+        if self.old_user(email):
+            userid = self.email_userid[email]
+            entry = self.find_val(userid)
+            return entry['password'] == password
+        else:
+            False
+        
+    def old_user(self,email):
+        
+        emails = list(self.email_userid.keys())
+        return email in emails
+        
+    def delete(self,userid):
+        
+        entry = self.find_val(userid)
+        super().delete(entry)
+        del self.id_index[userid]
+        for userid in range(userid+1 , len(self.list)+1):
+            self.id_index[userid] -= 1
+        
+    def update(self,userid,new_value,entity = 'email'):
```

Comment:
I don't get what this function does at all! What does the last argument, `entity`, do?

Reply:
it updates a key/value pair of  the inserted entry! entity tells us which key/value pair we are going to edit